Tasks/Common/VstsAzureHelpers_/InitializeAzModuleFunctions.ps1

```diff
@@ -135,8 +135,8 @@ function Initialize-AzSubscription {
 
     } elseif ($Endpoint.Auth.Scheme -eq 'ManagedServiceIdentity') {
         try {
-            Write-Host "##[command]Connect-AzAccount -Identity @processScope"
-            $null = Connect-AzAccount -Identity @processScope
+            Write-Host "##[command]Connect-AzAccount -Environment $environmentName -Identity @processScope"
```

Comment:
Does it work with Azure Cloud?

Reply:
It may be difficult for us to test this logic branch against AzureCloud internally, since our AzureCloud service connection is SPN-based. 

Is it sufficient to run this in a non-MSFT crafted ADO environment with `PowerShell@2` and an AzureCloud MSI Service Connection?

Reply:
Connect-AzAccount working for `-Environment AzureCloud` with self-hosted agent via MSI:

https://dev.azure.com/nina0881/old-reliable/_build/results?buildId=98&view=logs&j=cf7856eb-6d49-578a-94f5-e786517c8660&t=88b8ca6b-dadf-594c-d7f2-6b82965b601a&l=14